assets/js/googlesitekit/api/index.js

```diff
@@ -16,15 +16,125 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import md5 from 'md5';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
 /**
  * Internal dependencies
  */
-import { deleteItem, getKeys } from './cache';
-import { createCacheKey, siteKitRequest } from './index.private';
+import {
+	deleteItem,
+	getItem,
+	getKeys,
+	setItem,
+} from 'assets/js/googlesitekit/api/cache';
+import { sortObjectProperties } from 'assets/js/util';
 
 // Caching is enabled by default.
 let cachingEnabled = true;
 
+const KEY_SEPARATOR = '::';
+
+/**
+ * Create a cache key for a set of type/identifier/datapoint values.
+ *
+ * @private
+ * @param {string} type        The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier  The data identifier, eg. a module slug like `'search-console'`.
+ * @param {string} datapoint   The endpoint to request data from.
+ * @param {Object} queryParams Query params to send with the request.
+ * @return {string} The cache key to use for this set of values.
+ */
+export const createCacheKey = ( type, identifier, datapoint, queryParams = {} ) => {
+	const keySections = [ type, identifier, datapoint ].filter( ( keySection ) => {
+		return !! keySection && keySection.length;
+	} );
+
+	if (
+		keySections.length === 3 &&
+		( !! queryParams ) && ( queryParams.constructor === Object ) &&
+		Object.keys( queryParams ).length
+	) {
+		keySections.push(
+			md5( JSON.stringify( sortObjectProperties( queryParams ) ) )
+		);
+	}
+
+	return keySections.join( KEY_SEPARATOR );
+};
+
+/**
+ * Make a request to a WP REST API Site Kit endpoint.
+ *
+ * @private
+ * @param {string}  type                The data to access. One of 'core' or 'modules'.
+ * @param {string}  identifier          The data identifier, eg. a module slug like `'search-console'`.
+ * @param {string}  datapoint           The endpoint to request data from.
+ * @param {Object}  options             Options to pass to the request
```

Comment:
Optional parameters like this should start with `Optional.`. Also this is missing the period, every comment should have one (even if it's technically not a sentence).